arch/arm/core/aarch64/reset.S

```diff
@@ -65,39 +121,21 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	adr	x0, 1f
 	msr	elr_el3, x0
 	eret
-2:
-	/* Boot from EL2 not supported */
-	bl	.
-1:
 #endif
-	/* Setup vector table */
-	adr	x0, _vector_table
-
-	switch_el x1, 3f, 2f, 1f
-3:
-	/* Initialize VBAR */
-	msr	vbar_el3, x0
-
-	/* SError, IRQ and FIQ routing enablement in EL3 */
-	mrs	x0, scr_el3
-	orr	x0, x0, #(SCR_EL3_IRQ | SCR_EL3_FIQ | SCR_EL3_EA)
-	msr	scr_el3, x0
-
-	/* Disable access trapping in EL3 for NEON/FP */
-	msr	cptr_el3, xzr
-
 	/*
 	 * Enable the instruction cache, stack pointer and data access
-	 * alignment checks and disable speculative loads.
+	 * alignment checks.
 	 */
 	mov	x1, #(SCTLR_I_BIT | SCTLR_A_BIT | SCTLR_SA_BIT)
 	mrs	x0, sctlr_el3
 	orr	x0, x0, x1
 	msr	sctlr_el3, x0
+	isb
 	b	0f
+
 2:
 	/* Initialize VBAR */
-	msr	vbar_el2, x0
+	msr	vbar_el2, x19
```

Comment:
"disable speculative loads" can be removed from `sctlr_el2` setting as well. (@ line 150 - mentioning here as not able to put comment there). Although it's debatable if we want to support `EL2` at all. I think `EL1` and `EL3` support is good enough to avoid `EL2` intricacies.

Reply:
I will change the comment in 'EL2' section as well. We can not support Zephyr in EL3/EL2 as of now, so I ignore the EL2 part.